### PR TITLE
trezor-agent: add systemd service

### PR DESCRIPTION
**Pull request by jb55**
###### Motivation for this change

Option to enable a [trezor-agent](https://github.com/romanz/trezor-agent) service that allows you to use your Trezor as a GnuPG agent. This builds upon #28656.

Closes #28656 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/programs/gnupg.nix [hunk 1]
```diff
@@ -96,6 +118,19 @@ in
         message = "You can't use ssh-agent and GnuPG agent with SSH support enabled at the same time!";
       }
     ];
-  };
+  } // (mkIf cfg.trezor-agent.enable  {
+    systemd.user.services.trezor-agent = {
+      description = "Trezor GnuPG Agent";
+
+      wantedBy = [ "multi-user.target" ];
```
Comment (peterhoeg) [inline nixos/modules/programs/gnupg.nix]: multi-user.target isn't a valid target for user sessions.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -25802,18 +25802,26 @@ EOF
...
-    src = fetchPypi {
-      inherit pname version;
-      sha256 = "1i5cdamlf3c0ym600pjklij74p8ifj9cv7xrpnrfl1b8nkadswbz";
-    };
+    src = "${pkgs.fetchFromGitHub {
+      repo = "trezor-agent";
+      owner = "romanz";
+      rev = "v${version}";
+      sha256 = "1wrmf76016ksx394l1xknnqn08sc5pid5ihri619cd83zsxpkfvk";
+    }}/agents/trezor";
```
Comment (peterhoeg) [inline pkgs/top-level/python-packages.nix]: What's wrong with the pypi package?

**Comment by jb55** [comment]
> multi-user.target isn't a valid target for user sessions. 

ha I actually just remembered this before I went to sleep last night and
was going to fix it when I woke up (fixing now)

**Comment by jb55** [comment]
> What's wrong with the pypi package?

As far as I can tell it hasn't been uploaded yet...

https://pypi.python.org/pypi/trezor_agent/

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (jb55) [reply pkgs/top-level/python-packages.nix]: The author uploaded a new version, I'm going to update this PR this evening (-0700)

**Pull request closed without merging**